hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -742,6 +791,20 @@ protected boolean updateRecordWithExpiry(Data key, Object value, R record, Expir
                 disableWriteThrough, completionId, source, origin);
     }
 
+    protected void updateRecordWithExpiry(Data key, CacheRecord record, ExpiryPolicy expiryPolicy, long now, String source) {
+        expiryPolicy = getExpiryPolicy(record, expiryPolicy);
+        long expiryTime = CacheRecord.TIME_NOT_AVAILABLE;
+        try {
+            Duration expiryDuration = expiryPolicy.getExpiryForUpdate();
+            if (expiryDuration != null) {
+                expiryTime = getAdjustedExpireTime(expiryDuration, now);
+            }
+        } catch (Exception e) {
+            ignore(e);
```

Comment:
what was the reason that we ignored this exception?

Reply:
ExpiryPolicy#getExpiryForUpdate doc says
```
   * Should an exception occur while determining the Duration, an implementation
   * specific default Duration will be used.
```
That is why we ignore the exception. There is another `updateRecordWithExpiry` method that is used by put operations.